atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/ranges/RangeBoundPredicates.java

```diff
@@ -109,5 +109,23 @@ public Builder columnSelection(Collection<byte[]> columns) {
         public RangeBoundPredicates build() {
             return new RangeBoundPredicates(predicates.toString(), args);
         }
+
+        public Builder startCellTsInclusiveOracle(byte[] startRowInclusive, byte[] startColumnInclusive,
+                Long startTsInclusive) {
+            if (startTsInclusive != null) {
+                Preconditions.checkArgument(startRowInclusive.length > 0);
+                Preconditions.checkArgument(startColumnInclusive.length > 0);
+                // Warning: this syntax is not supported by Oracle
+                predicates.append(reverse
+                        ? " AND row_name <= ? AND col_name <= ? AND ts <= ? "
+                        : " AND row_name >= ? AND col_name >= ? AND ts >= ? ");
```

Comment:
This is incorrect. This should be like `(row_name = ? AND col_name >= ?) OR row_name > ?`, but unfortunately that doesn't perform well on Oracle.

Also `ts <= ?` in both cases